tests/js/gamepad-mock.js

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright (c) 2020 The Gamepad Navigator Authors
+See the AUTHORS.md file at the top-level directory of this distribution and at
+https://github.com/fluid-lab/gamepad-navigator/raw/master/AUTHORS.md.
+
+Licensed under the BSD 3-Clause License. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the BSD 3-Clause License at
+https://github.com/fluid-lab/gamepad-navigator/blob/master/LICENSE
+*/
+
+/* global gamepad */
+
+(function (fluid) {
+    "use strict";
+
+    fluid.registerNamespace("gamepad.tests.utils");
+
+    /**
+     *
+     * Generate a mock of the standard Gamepad object. Optionally, it can take the
+     * inputSpec as its parameter. The inputSpec should be in the following formats:
+     *
+     * {
+     *   buttons: { "2": 1 }
+     * }
+     *
+     * The above inputSpec will return a gamepad mock with the value of button 2 to 1.
+     *
+     * {
+     *   axes: { "1": 0.753 }
+     * }
+     *
+     * The above inputSpec will return a gamepad mock with the value of axes 2 to 0.753.
+     *
+     * {
+     *   buttons: {
+     *     "5": 1,
+     *     "11": 1
+     *   },
+     *   axes: { "3": 0.92 }
+     * }
+     *
+     * The above inputSpec will return a gamepad mock with the value of button 5 to 1,
+     * button 11 to 1, and of axes 3 to 0.92.
+     *
+     * @param {Object} inputSpec - (optional) The object containing the desired value for
+     *                             the gamepad inputs.
+     * @return {Array} - The array containing the mock gamepad mock object.
+     *
+     */
+    gamepad.tests.utils.generateGamepadMock = function (inputSpec) {
```

Comment:
I'm a little worried that this approach will not be sufficient, it seems to me like you will often need to trigger a change of button state rather than initialising the mock with a particular state.

Reply:
This function is just an all-purpose mock generator for use in all types of tests in the future. We are using other functions in our tests, which use this function for generating a gamepad mock with varying states (such as change in button values, axes values, etc.).

Reply:
We can certainly give it a try and change the approach if it becomes burdensome.